### PR TITLE
Adds method to generate extended public key

### PR DESCRIPTION
**Pull request by dchatzieleftheriou-bc**
This adds method to generate a extended public keys

### CODE REVIEW

File: Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift [hunk 1]
```diff
@@ -254,6 +276,15 @@ extension PrivateKey {
         return encoded
     }
 
+    /// Adds SHA256 twice to the given data
+    /// - Parameter data: A `Data` object to be processed
+    /// - Returns: A processed `Data` object
+    private func addChecksum(data: Data) -> Data {
+        var data = data
+        data += data.doubleSHA256.prefix(4)
+        return data
```
Comment (paulo-bc) [inline Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift]:
```suggestion
        data + data.doubleSHA256.prefix(4)
```

File: Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift [hunk 2]
```diff
@@ -232,20 +233,41 @@ extension PrivateKey {
...
+        let checksummed = addChecksum(data: serialized)
+        let encoded = Base58.encode(checksummed)
+        return encoded
+    }
+
+    /// Provides the extended public key
+    public func extendedPublic() -> String {
+
+        func serialize() -> Data {
+            var bytes = [UInt8]()
```
Comment (paulo-bc) [inline Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift]: Could we add here the same comments that we have `func serialize() -> Data`? Will make it easier to read.

File: Sources/MetadataHDWalletKit/Wallet/Coin.swift [hunk 3]
```diff
@@ -33,6 +33,23 @@ public enum Coin {
         }
     }
 
+    // https://github.com/satoshilabs/slips/blob/master/slip-0132.md
+    public var publicKeyVersion: UInt32 {
+        switch self {
+        case .litecoin:
+            return 0x019da462
+        case .bitcoinCash: fallthrough
```
Comment (paulo-bc) [inline Sources/MetadataHDWalletKit/Wallet/Coin.swift:41]: `fallthrough`?  I thin adding a `fatalError("Not implemented")` and a TODO here would be better suited.

Comment (dchatzieleftheriou-bc) [reply Sources/MetadataHDWalletKit/Wallet/Coin.swift:41]: `fallthough` to get the same value as `bitcoin`

File: Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift (hunk 2 above)
Comment (dchatzieleftheriou-bc) [reply Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift]: yeah i could add the "same" comments

**Review by jackpooleybc — APPROVED** [review]

**Pull request merged**